core/trino-main/src/main/java/io/trino/execution/buffer/SpoolingExchangeOutputBuffer.java

```diff
@@ -57,6 +58,7 @@ public SpoolingExchangeOutputBuffer(
         this.stateMachine = requireNonNull(stateMachine, "stateMachine is null");
         this.outputBuffers = requireNonNull(outputBuffers, "outputBuffers is null");
         checkArgument(outputBuffers.getType() == SPOOL, "Expected a SPOOL output buffer");
+        // this assignment is expected to be followed by an assignment of a final field to ensure safe publication
```

Comment:
If the object created is published in not thread safe manner it may be published partially constructed, as the assignments are free to be reordered by the CPU. Assigning of a `final` field provides memory ordering guarantees. The `SpoolingExchangeOutputBuffer` is in fact published under a lock in a thread safe way, as the `delegate` variable is declared `volatile`: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/buffer/LazyOutputBuffer.java#L64. Although if the `delegate` accesses ever become non volatile and the order of initialization in this constructor is changed it might introduce a hard to debug concurrency bug. Thus I though that it deserves an explicit comment.